code/__HELPERS/lists.dm

```diff
@@ -53,10 +53,16 @@
 
 //Checks for specific types in a list
 /proc/is_type_in_list(atom/A, list/L)
+	if (!L.len)
+		return 0
+	if (!isnum(L[L[1]]))
+		generate_type_list_cache(L)

```

Comment:
@optimumtact yes that was my original "solution", but since I got SO CLOSE with an automated solution I feel it's probably for the best to get one to work.